lib/bap/bap_log.ml

```diff
@@ -54,13 +54,26 @@ let print_message ppf msg =
   | Error -> eprintf "%s@\n%!" msg.message
   | _ -> ()
 
+let lock () =
+  let file = Filename.get_temp_dir_name () / "bap_lock" in
```

Comment:
1) `bap.lock` is a much more conventional name 
2) this lock file would be global to all users, so my suggestion is to encode in the name of the lock file the path to the folder, that you're trying to create, salted with bap-prefix, the simplest idea is to take the digest of the path, e.g, 
```ocaml
let lock_filename logdir = 
    let digest = Md5.digest_string logdir in
    let name = sprintf "bap-%s.lock" (Md5.to_hex digest) in
    Filename.get_temp_dir_name () / name
```